packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -129,20 +129,53 @@ storage:
       contents:
         inline: |
           #!/bin/bash -xe
+          # A comma-separated list of major device numbers. Modify to control which device types
+          # are considered for data RAID.
+          # https://www.kernel.org/doc/Documentation/admin-guide/devices.txt
+          major_numbers="8,259"
 
-          function persist_data_raid() {
-            local raid_config=$(mdadm --detail --scan)
-            echo "$${raid_config}" | tee -a /etc/mdadm.conf
-            # Don't mount if no RAID is configured.
-            [ -z "$${raid_config}" ] && return 0
```

Comment:
Will this work if you reboot the node? Won't it just create a new raid array and format it on reboot?

Reply:
The systemd service is only executed if the mdadm config file does not exist.

Reply:
Oh, good point. But this will erase all the data and create a new fs if, for some reason, the mdadm.conf file is deleted. Or it will fail in some way if the mdadm.conf file is not there, because the raid array already exists in any case....

What if we check if it exists and exit in that case (creating the mdadm.conf) as it was doing before?

Reply:
Ok, I've added a check to do nothing if the config file is missing but a RAID was configured.

Reply:
Thanks!